sandbox/modules/identity/src/main/java/org/opensearch/identity/authz/PermissionFactory.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.authz;
+
+/**
+ * This class is used to create Permission instances. The factory can create both standard Permissions which have specific
+ * formatting requirements and legacy permissions which are not checked for validity on creation.
+ */
+public class PermissionFactory {
+
+    public final static String[] INVALID_CHARACTERS = new String[] { ":", "" }; // This is a placeholder for what may want to be banned
```

Comment:
making private

If you made it public so tests can access, then just remote the public keyword so its internal